### PR TITLE
Small cleanups after TF 0.12 migration

### PR DESCRIPTION
**Pull request by bacek**
1. Remove some unused code.
2. Add `-` as allowed variable name character

### CODE REVIEW

File: core/core.ts [hunk 1]
```diff
@@ -695,7 +675,7 @@ export function fileGenerator(): FileGenerator {
   };
 }
 
-const VAR_REGEX = /\${([a-zA-Z0-9_.]+)}$/;
+const VAR_REGEX = /\${([a-zA-Z0-9-_.]+)}$/;
```
Comment (timbod7) [inline core/core.ts:678]: Do we actually have minus signs in resource names?

File: core/core.ts [hunk 2]
```diff
@@ -576,38 +576,18 @@ export function fileGenerator(): FileGenerator {
...
-      return hcl2.stringLit(t);
-    }
-  }
-
-
   function generateMetadata(resource: ResourceDetails): hcl2.BodyItem[] {
     const items: hcl2.BodyItem[] = [];
 
-    {
     const lifecycle_items: hcl2.BodyItem[] = [];
```
Comment (timbod7) [inline core/core.ts:582]: I prefer it as it was, where `lifecycle_items` was only scoped as required, rather scoped to the end of the function?

File: core/core.ts (hunk 1 above)
Comment (bacek) [reply core/core.ts:678]: Yes. `zone_id` is the biggest offender on our side.

File: core/core.ts (hunk 2 above)
Comment (bacek) [reply core/core.ts:582]: Will fix it

Comment (timbod7) [reply core/core.ts:582]: I don't really care - I'm just pointing out it was written as is intentionally.

**Review by timbod7 — APPROVED** [review]

**Pull request merged**